fdbserver/worker.actor.cpp

```diff
@@ -431,16 +432,23 @@ void updateCpuProfiler(ProfilerRequest req) {
 			break;
 		}
 		break;
+	case ProfilerRequest::Type::GPROF_HEAP:
+#if defined(__linux__) && defined(USE_GPERFTOOLS) && !defined(VALGRIND)
+		HeapProfilerDump("User triggered heap dump");
+#endif
+		break;
 	}
 }
 
 ACTOR Future<Void> runProfiler(ProfilerRequest req) {
 	if (req.action == ProfilerRequest::Action::RUN) {
 		req.action = ProfilerRequest::Action::ENABLE;
 		updateCpuProfiler(req);
-		wait(delay(req.duration));
-		req.action = ProfilerRequest::Action::DISABLE;
-		updateCpuProfiler(req);
+		if (req.type != ProfilerRequest::Type::GPROF_HEAP) {
```

Comment:
Can you refactor this a bit, as RUN is now being used to mean two different things.  I think maybe the cleaner way to do this would be to make updateCpuProfiler CPU profiling only, and to do the enable -> wait -> disable flow internally and returns a future that we wait on here, and then handle heap profiling separately?